AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -355,7 +355,10 @@ - (id)initWithCoder:(NSCoder *)decoder {
         _effectiveDate = [decoder decodeInt64ForKey:@"effectiveDate"];
         _location = [decoder decodeObjectOfClass:[AWSPinpointEndpointProfileLocation class] forKey:@"location"];
         _demographic = [decoder decodeObjectOfClass:[AWSPinpointEndpointProfileDemographic class] forKey:@"demographic"];
-        _attributes = [decoder decodeObjectOfClass:[NSDictionary class] forKey:@"attributes"];
+
+        NSSet * attributesClasses = [NSSet setWithObjects:[NSDictionary class], [NSArray class], [NSString class], nil];
+        _attributes = [decoder decodeObjectOfClasses:attributesClasses forKey:@"attributes"];
+
         _metrics = [decoder decodeObjectOfClass:[NSDictionary class] forKey:@"metrics"];
```

Comment:
Should we do this for other `NSDictionary`s ?

Reply:
I've pushed an update which includes tests for the metrics property and updates the `initWithCoder` function so it uses the `decodeObjectOfClasses` method. It did not fail the tests before making this change, but it is the safer option since it is also a nested type.